### PR TITLE
[SPARK-47579][CORE][PART4] Migrate logInfo with variables to structured logging framework

### PR DESCRIPTION
**Pull request by zeotuan**
The PR aims to migrate `logInfo` in Core module with variables to structured logging framework.

### Why are the changes needed?

To enhance Apache Spark's logging system by implementing structured logging.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by zeotuan** [comment]
@gengliangwang Please help review this. I will merge this after #46739

**Comment by gengliangwang** [comment]
@zeotuan I was aware of this PR when I worked on https://github.com/apache/spark/pull/46739, since you were not responding to the jira comment.

Please resolve the merge conflicts and I will review and merge this one.

**Comment by zeotuan** [comment]
@gengliangwang Hi please help review

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 1]
```diff
@@ -219,7 +221,7 @@ private[spark] class Executor(
         if (sessionBasedRoot.isDirectory && sessionBasedRoot.exists()) {
           Utils.deleteRecursively(sessionBasedRoot)
         }
-        logInfo(s"Session evicted: ${state.sessionUUID}")
+        logInfo(log"Session evicted: ${LogMDC(UUID, state.sessionUUID)}")
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala:224]: SESSION_ID

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 2]
```diff
@@ -1078,8 +1086,9 @@ private[spark] class Executor(
 
   private def createClassLoader(urls: Array[URL], useStub: Boolean): MutableURLClassLoader = {
     logInfo(
-      s"Starting executor with user classpath (userClassPathFirst = $userClassPathFirst): " +
-      urls.mkString("'", ",", "'")
+      log"Starting executor with user classpath" +
+        log" (userClassPathFirst = ${LogMDC(CLASS_PATH, userClassPathFirst)}): " +
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala:1090]: userClassPathFirst is a boolean

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 3]
```diff
@@ -1123,12 +1132,13 @@ private[spark] class Executor(
...
     val classLoader = if (classUri != null) {
-      logInfo("Using REPL class URI: " + classUri)
+      logInfo(log"Using REPL class URI: ${LogMDC(LogKeys.URI, classUri)}")
       new ExecutorClassLoader(conf, env, classUri, parent, userClassPathFirst)
     } else {
       parent
     }
-    logInfo(s"Created or updated repl class loader $classLoader for $sessionUUID.")
+    logInfo(log"Created or updated repl class loader ${LogMDC(CLASS_LOADER, classLoader)}" +
+      log" for ${LogMDC(UUID, sessionUUID)}.")
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala:1141]: SESSION_ID

File: core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala [hunk 4]
```diff
@@ -98,10 +98,11 @@ object SparkHadoopWriter extends Logging {
           iterator = iter)
       })
 
-      logInfo(s"Start to commit write Job ${jobContext.getJobID}.")
+      logInfo(log"Start to commit write Job ${MDC(JOB_ID, jobContext.getJobID)}.")
       val (_, duration) = Utils
         .timeTakenMs { committer.commitJob(jobContext, ret.toImmutableArraySeq) }
-      logInfo(s"Write Job ${jobContext.getJobID} committed. Elapsed time: $duration ms.")
+      logInfo(log"Write Job ${MDC(JOB_ID, jobContext.getJobID)} committed." +
+        log" Elapsed time: ${MDC(TOTAL_TIME, duration)} ms.")
```
Comment (gengliangwang) [inline core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala:105]: `DURATION`, or we can just merge `DURATION` and `TOTAL_TIME`

**Comment by gengliangwang** [comment]
LGTM except for a few minor comments. Thanks for the work!

File: common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala [hunk 5]
```diff
@@ -764,6 +800,7 @@ object LogKeys {
   case object WORKER extends LogKey
   case object WORKER_HOST extends LogKey
   case object WORKER_ID extends LogKey
+  case object WORKER_MEMORY extends LogKey
```
Comment (gengliangwang) [inline common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:803]: This is not used.

File: common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala [hunk 6]
```diff
@@ -727,6 +760,8 @@ object LogKeys {
   case object TOPIC_PARTITION_OFFSET_RANGE extends LogKey
   case object TOTAL extends LogKey
   case object TOTAL_EFFECTIVE_TIME extends LogKey
+  case object TOTAL_RECORDS_READ extends LogKey
```
Comment (gengliangwang) [inline common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:763]: This is not used.

Comment (gengliangwang) [reply common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:763]: This is not used.

**Comment by gengliangwang** [comment]
@zeotuan please fix the test failures and remove all the unused keys(I didn't go over all the new keys).
I will merge this one after that.

File: common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala [hunk 7]
```diff
@@ -373,6 +382,7 @@ object LogKeys {
   case object NEW_VALUE extends LogKey
   case object NEXT_RENEWAL_TIME extends LogKey
   case object NODES extends LogKey
+  case object NODE_IDS extends LogKey
```
Comment (gengliangwang) [inline common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala:385]:
```suggestion
```

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**